### PR TITLE
CSHARP-4347: Add log messages to Server selection spec

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Logging/StructuredLogTemplateProvidersCluster.cs [hunk 1]
```diff
@@ -26,29 +26,74 @@ internal static partial class StructuredLogTemplateProviders
...
+                ServerSelectionCommonParams(Failure),
+                (e, _) => GetParams(
+                    e.ClusterId,
+                    e.ServerSelector.ToString(),
+                    e.OperationId,
+                    e.OperationName,
+                    e.ClusterDescription.ToString(),
+                    "Server selection failed",
+                    "exception123"));
+                    //e.Exception?.ToString()));
```
Comment (JamesKovacs) [inline src/MongoDB.Driver.Core/Core/Logging/StructuredLogTemplateProvidersCluster.cs:91]: Debug code `exception123` should be replaced with commented impl on line 91.

File: src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs [hunk 2]
```diff
@@ -102,6 +103,8 @@ public async Task<bool> ExecuteAsync(IReadBinding binding, CancellationToken can
         }
 
         // private methods
+        private IDisposable BeginOperation() => EventContext.BeginOperation(null, "listDatabases");
+
```
Comment (JamesKovacs) [inline src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs:107]: Missing calls to `BeginOperation` in `Execute` and `ExecuteAsync`.

File: src/MongoDB.Driver.Core/ReadPreference.cs [hunk 3]
```diff
@@ -115,6 +115,7 @@ public static ReadPreference FromBsonDocument(BsonDocument document)
                         break;
 
                     case "tags":
+                    case "tagSets":
```
Comment (JamesKovacs) [inline src/MongoDB.Driver.Core/ReadPreference.cs:118]: This looks unrelated. Move to another ticket?

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Feedback and questions for your consideration.

File: src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs (hunk 2 above)
Comment (JamesKovacs) [reply src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs:107]: Above comment needs addressing.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
One comment was missed.

File: src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Operations/DatabaseExistsOperation.cs]: Thanks, `BeginOperation` removed.

**Pull request merged**